### ISSUE TITLE
Autodetect query object

### ISSUE THREAD

**Issue by burkekevinm** [omniphx/adminite#42]
If no object is entered you could use a regular expression on the query content to find the last instance of from (to avoid subqueries in the earlier part of the query) and then regular expression the next full word you could see if that matches the list and auto-detect if one wasn't specified in the query and update.

**Comment by omniphx** [comment]
Not sure I follow, are requesting auto-completion in subqueries?

**Comment by burkekevinm** [comment]
So on the query window you have the dropdown to select which object you are
querying against (and to support the add fields). I'm saying you could
check the query itself to populate that if it isn't populated.

What I'm finding is I start writing a query and don't select the object and
then realize I don't remember a field name so I go up and select the object
then go to fields to figure it out. You could use a regular expression
similar to from(?!.*from)\s(\w+) to detect the last instance of from and
the word that follows it. https://regex101.com/r/00QS5I/1

>

**Comment by omniphx** [comment]
Ah gotcha, I can make that happen pretty easily

**Comment by burkekevinm** [comment]
Yeah I figured it would be pretty straight forward. Do you want me to add
something on the issues?

>

**Comment by omniphx** [comment]
One issue with a dynamically setting the object context is that it would prevent a user from overriding the auto-correct in situations where you are adding fields in a subquery or parent relationship.

Eventually, I'll make the auto-correct more robust and dynamically setting the sobject context will be a non-issue.
